types/options.go

```diff
@@ -481,33 +481,6 @@ func ReloadConfigurationFile(configFile string, storeOptions *StoreOptions) erro
 	if config.Storage.Options.MountOpt != "" {
 		storeOptions.GraphDriverOptions = append(storeOptions.GraphDriverOptions, fmt.Sprintf("%s.mountopt=%s", config.Storage.Driver, config.Storage.Options.MountOpt))
 	}
-
-	uidmap, err := idtools.ParseIDMap([]string{config.Storage.Options.RemapUIDs}, "remap-uids")
-	if err != nil {
-		return err
-	}
-	gidmap, err := idtools.ParseIDMap([]string{config.Storage.Options.RemapGIDs}, "remap-gids")
-	if err != nil {
-		return err
-	}
-
-	if config.Storage.Options.RemapUser != "" && config.Storage.Options.RemapGroup == "" {
-		config.Storage.Options.RemapGroup = config.Storage.Options.RemapUser
-	}
-	if config.Storage.Options.RemapGroup != "" && config.Storage.Options.RemapUser == "" {
-		config.Storage.Options.RemapUser = config.Storage.Options.RemapGroup
-	}
-	if config.Storage.Options.RemapUser != "" && config.Storage.Options.RemapGroup != "" {
-		mappings, err := idtools.NewIDMappings(config.Storage.Options.RemapUser, config.Storage.Options.RemapGroup)
-		if err != nil {
-			logrus.Warningf("Error initializing ID mappings for %s:%s %v\n", config.Storage.Options.RemapUser, config.Storage.Options.RemapGroup, err)
-			return err
-		}
-		uidmap = mappings.UIDs()
-		gidmap = mappings.GIDs()
-	}
-	storeOptions.UIDMap = uidmap
-	storeOptions.GIDMap = gidmap
```

Comment:
So the `StoreOptions.[UG]IDMap` fields still exist; they still affect `CreateContainer` and `putLayer`; and they, AFAICS, are potentially passed by Podman and potentially? user-configurable in `podman/pkg/domain/infra`. Yet the c/storage-level option has been removed.

What’s the full story here? Are the fields _also_ intended to be irrelevant / removed, and that just didn’t happen in order not to break Podman builds? Have the fields been used for two different purposes, so that we are now using only one? Are we changing what the fields mean now?

Reply:
The StoreOptions.[UGMap] are still honored.  I've changed them to not be read from the `storage.conf` file and to not affect the `Init()` call for each driver.   I don't think the internal directories used by each driver must be affected by the mappings.  I don't see a valid use case for that, unless we are planning to run without being root (either on the host, or in a user namespace).

```
# bin/podman run --rm -it --subuidname containers alpine sh -c 'grep . /proc/self/*id_map; ls -l /'
/proc/self/gid_map:         0     200000  268435456
/proc/self/uid_map:         0     200000  268435456
total 12
drwxr-xr-x    2 root     root          4096 Jun 18 14:16 bin
drwxr-xr-x    5 root     root           360 Jun 25 09:48 dev
drwxr-xr-x    1 root     root            25 Jun 25 09:48 etc
drwxr-xr-x    2 root     root             6 Jun 18 14:16 home
drwxr-xr-x    6 root     root          4096 Jun 18 14:16 lib
drwxr-xr-x    5 root     root            44 Jun 18 14:16 media
drwxr-xr-x    2 root     root             6 Jun 18 14:16 mnt
drwxr-xr-x    2 root     root             6 Jun 18 14:16 opt
dr-xr-xr-x  577 nobody   nobody           0 Jun 25 09:48 proc
drwx------    2 root     root             6 Jun 18 14:16 root
drwxr-xr-x    1 root     root            42 Jun 25 09:48 run
drwxr-xr-x    2 root     root          4096 Jun 18 14:16 sbin
drwxr-xr-x    2 root     root             6 Jun 18 14:16 srv
dr-xr-xr-x   13 nobody   nobody           0 Jun 23 19:41 sys
drwxrwxrwt    2 root     root             6 Jun 18 14:16 tmp
drwxr-xr-x    7 root     root            66 Jun 18 14:16 usr
drwxr-xr-x   12 root     root           137 Jun 18 14:16 var
```

Reply:
ACK. Reading the documentation of the fields, they do seem to have been documented for two different purposes.

> I don't think the internal directories used by each driver must be affected by the mappings.

I agree.